File: Solution_42583.cs
using System;
using System.Collections;
using System.Collections.Generic;

using System.Linq;

public class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;
        List<int> lTruck = new List<int>();
        
        int nIdx = 0;
        int nChkWeight = 0;
        answer = bridge_length;
        
        while(nIdx < truck_weights.Length){
            lTruck.Add(truck_weights[nIdx]);
            
            nChkWeight = lTruck.Sum();
            
            if(nIdx + 1 <truck_weights.Length ){
                if(nChkWeight + truck_weights[nIdx + 1] > weight){
                    answer+=2;
                    lTruck.RemoveAt(0);
                }else answer++;
            }else{
                answer++;
            }
            nIdx++;
        }
        
        return answer;
    }
}

File: Solution_42583.js
function solution(bridge_length, weight, truck_weights) {
    var answer = 0;
    let truck = [...truck_weights];
    let finish = [];
    let passing = [[0, 0]];
    let weightOnBridge = 0;
    let dupCnt = 0;
    while (passing.length || truck.length) {
        if (passing[0][1] === answer) weightOnBridge -= passing.shift()[0];
        // console.log('w eightOnBridge', answer, weightOnBridge)
        if (weightOnBridge + truck[0] <= weight) {
            weightOnBridge += truck[0];
            passing.push([truck.shift(), answer + bridge_length]);
        } else {
            if (passing[0]) answer = passing[0][1] - 1;
        }
        answer++;
    }
    console.log(passing, answer);
    return answer;
}
